#28 Can't read repository name from $PWD

mohamednizar opened the pull request:
#28 
Now it can read the root file name. 

FIx the $PWD read issue associated with gopwd package. https://github.com/g4s8/gopwd.

See the out put of test cases:

New unit test added for confirm the reponame from working directory.

This was stated in https://github.com/g4s8/gopwd/issues/4 and fixed
```
:~/gopwd$ go run main.go 
gopwd <nil>

```

0crat commented:
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this pull request

0crat commented:
@mohamednizar[/z](https://www.0crat.com/u/mohamednizar) this pull request is too small, just 1 lines changed (less than 10), there will be no formal code review, see [§53](http://www.zerocracy.com/policy.html#53) and [§28](http://www.zerocracy.com/policy.html#28); in the future, try to make sure your pull requests are not too small; @g4s8[/z](https://www.0crat.com/u/g4s8) please review this and merge or reject

<!-- https://www.0crat.com/footprint/CF7JL4282/3281d7f6-7980-4e15-9e31-beab4b387bfb, version: 0.50.34, hash: ${buildNumber} -->

g4s8 commented:
@mohamednizar please check #33 and merge master branch

g4s8 commented on gitstrap.go:
```diff
@@ -18,13 +17,22 @@ import (
 )
 
+var (
+    _, b, _, _ = runtime.Caller(0)
```
@mohamednizar it won't work, since `caller` doesn't return current directory:
> Caller reports file and line number information about function invocations on the calling goroutine's stack. The argument skip is the number of stack frames to ascend, with 0 identifying the caller of Caller. (For historical reasons the meaning of skip differs between Caller and Callers.) The return values report the program counter, file name, and line number within the file of the corresponding call. The boolean ok is false if it was not possible to recover the information. 

E.g. if I compile go programm:
```go
package main

import (
        "fmt"
        "runtime"
)

func main() {
        _, b, _, _ := runtime.Caller(0)
        fmt.Println(b)
}
```
at `/tmp/1` location, and then start it from different directory it'll print `/tmp/1/` file location:
```bash
$ cd ~
$ /tmp/1/prog
/tmp/1/prog.go
$ pwd
/home/g4s8
```

mohamednizar replied:
FIx the $PWD read issue associated with gopwd package. Pointed the folk https://github.com/mohamednizar/gopwd as well as created a pull request for https://github.com/g4s8/gopwd.

See the out put of test cases:

```
nizar@nizar-GL63-8SD:~/gopwd$ go run main.go 
gopwd <nil>

nizar@nizar-GL63-8SD:~$ go run gopwd/main.go 
nizar <nil>
```
Hope this would fix the issue #2

mohamednizar commented:
@g4s8 , I have change the code according to your guidelines, also added created a PR on https://github.com/g4s8/gopwd/pull/2 related to this. I have replaced the go library with my forlk, Once you merged that I can change it to the upstream.

g4s8 commented:
@mohamednizar please see my comment: https://github.com/g4s8/gitstrap/pull/32#issuecomment-548523791

Forking dependencies is not a correct way to solve the problem. The correct way will be creating a bug in dependency repo or submitting a pull request to fix it, then wait it to be solved and update the dependency.

mohamednizar commented:
> @mohamednizar please see my comment: [#32 (comment)](https://github.com/g4s8/gitstrap/pull/32#issuecomment-548523791)
> 
> Forking dependencies is not a correct way to solve the problem. The correct way will be creating a bug in dependency repo or submitting a pull request to fix it, then wait it to be solved and update the dependency.

@g4s8 If you noticed, I have created a pull request to the dependency. https://github.com/g4s8/gopwd/pull/2

mohamednizar commented:
@rultor merge

rultor commented:
> @rultor merge

@mohamednizar Thanks for your request. @g4s8 Please confirm this.

g4s8 commented:
@rultor merge

rultor commented:
> @rultor merge

@g4s8 OK, I'll try to merge now. You can check the progress of the merge [here](https://www.rultor.com/t/20837-552174152)

rultor commented:
> @rultor merge

@g4s8 Done! FYI, the full log is [here](https://www.rultor.com/t/20837-552174152) (took me 2min)

0crat commented:
Job `gh:g4s8/gitstrap#32` is not assigned, can't get performer

<!-- https://www.0crat.com/footprint/CF7JL4282/c0b10be9-7f91-4965-a3bf-eb18e7d5c444, version: 0.30.7, hash: 4a2fe9940 -->

0crat commented:
This job is not in scope

<!-- https://www.0crat.com/footprint/CF7JL4282/5a0d75dd-cdc2-4768-b6c8-8b6c067b3ffc, version: 0.30.7, hash: 4a2fe9940 -->
